src/app/config/constants/constants.ts

```diff
@@ -1,26 +1,35 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
 
 import apiURL from 'src/app/config/constants/apiURL';
+
 interface externalNameResponseFormat {
-  externalName: String;
+  externalName: string;
 };
 
-@Injectable({
-  providedIn: 'root',
-})
+@Injectable({ providedIn: 'root' })
 export class DoubtfireConstants {
-  constructor(private httpClient: HttpClient) {
-    httpClient.get(`${this.apiURL}/settings`).toPromise().then((response: externalNameResponseFormat) => {
-      this.externalName = response.externalName;
-    })
+  constructor(private http: HttpClient) {
+    this.loadExternalName();
   }
   public mainContributors: string[] = [
     'macite',              // Andrew Cain
     'alexcu',              // Alex Cummaudo
     'jakerenzella'         // Jake Renzella
   ];
-  public apiURL = apiURL;
-  public externalName: String = "Loading...";
-}
 
+  public loaded: boolean = false;
+  public apiURL: string = apiURL;
+
+  public getExternalName = new BehaviorSubject<string>('Loading...');
+
+  private loadExternalName() {
+    this.http.get<externalNameResponseFormat>(`${this.apiURL}/settings`)
+      .subscribe(result => this.getExternalName.next(result.externalName)),
```

Comment:
Definitely need a rename... the above does not read well

Reply:
I have reformatted, but can't think of a better name than `loadExternalName`

```
  // publish update to ExternalName when get request finishes.
  private loadExternalName() {
    const settingsUrl: string = `${this.apiURL}/settings`;

    this.http.get<externalNameResponseFormat>(settingsUrl)
      .subscribe(
        result => this.ExternalName.next(result.externalName)
      ),
      error => console.error(error);
  }
```

* getExternalName is bad because it doesn't return anything.
* maybe updateExternalName?

Reply:
Load was fine... it was the public getExternalName.

Reply:
👍

Reply:
this.ExternalName.next would be fine...